src/types.rs

```diff
@@ -149,19 +148,24 @@ impl<'a> ObjectName<'a> {
         for (i, c) in name.chars().enumerate() {
             match c {
                 // Object names cannot contain Carriage Return or Line Feed characters.
-                '\r' | '\n' => {}
                 // Avoid using "#" in your object names: gsutil interprets object names ending
                 // with #<numeric string> as version identifiers, so including "#" in object names
                 // can make it difficult or impossible to perform operations on such versioned
                 // objects using gsutil (see Object Versioning and Concurrency Control).
                 // Avoid using "[", "]", "*", or "?" in your object names:  gsutil interprets
                 // these characters as wildcards, so including them in object names can make
                 // it difficult or impossible to perform wildcard operations using gsutil.
-                '#' | '[' | ']' | '*' | '?' => {}
+                '\r'
+                | '\n'
+                | '#'
+                | '['
+                | ']'
+                | '*'
+                | '?'
+                | '\u{7F}'..='\u{84}'
+                | '\u{86}'..='\u{9F}' => {}
                 // Avoid using control characters that are illegal in XML 1.0 (#x7F–#x84 and #x86–#x9F):
                 // these characters will cause XML listing issues when you try to list your objects.
-                '\u{7F}'..='\u{84}' => {}
-                '\u{86}'..='\u{9F}' => {}
```

Comment:
I'd like the warning to be ignored here, each section has its own preceding comment about why they are not allowed, this loses that.